numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -932,6 +938,8 @@ typedef int (PyArray_FinalizeFunc)(PyArrayObject *, PyObject *);
 #define NPY_ARRAY_UPDATEIFCOPY    0x1000 /* Deprecated in 1.14 */
 #define NPY_ARRAY_WRITEBACKIFCOPY 0x2000
 
+#define _NPY_ARRAY_ENSURENOCOPY 0x4000
```

Comment:
These are flags that can be used in the "public" C-API. Why does this start with a '_'?

Reply:
I think it needs a comment like the other ones:
> This flag may be requested in constructor functions.

Reply:
cc: @seberg 

Earlier review comment which lead to underscore in the name, https://github.com/numpy/numpy/pull/19173#discussion_r703602707

Reply:
> Should this have an underscore at least, especially since right now I don't think we actually have support for it everywhere in the C-API? (@seberg)

Where is it not supported? I think it is strange to have a flag only accessible from the C-API and mark it with an underscore. What does the underscore signify?

Reply:
Yeah, I think on my first look, I thought it was at least skipped for `PyArray_FromAny`.  But it also sounds like we have to fix it anyway.  I suppose the only argument would be that it is currently "private" on the Python side as well.

I am happy with reversing this: though if we put this in, it seems very unlikely we want to back it out again (even if they way we expose it to python may change).